addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/NeatoHandlerFactory.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neato.internal;
+
+import static org.openhab.binding.neato.NeatoBindingConstants.THING_TYPE_VACUUMCLEANER;
+
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.neato.handler.NeatoHandler;
+import org.osgi.service.component.ComponentContext;
+
+/**
+ * The {@link NeatoHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Patrik Wimnell - Initial contribution
+ */
+public class NeatoHandlerFactory extends BaseThingHandlerFactory {
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_VACUUMCLEANER);
+
+    private static String email;
+    private static String password;
+
+    public static String getEmail() {
+        return email;
+    }
+
+    public static String getPassword() {
+        return password;
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected ThingHandler createHandler(Thing thing) {
+
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(THING_TYPE_VACUUMCLEANER)) {
+            return new NeatoHandler(thing);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+
+        Dictionary<String, Object> properties = componentContext.getProperties();
```

Comment:
This feels strange, the email/username is either part of the generic config or of the bridge config but I would not expect it here, can you explain why it should be here?

Reply:
Most times I see a reference to a / the bridge be passes to the discovery service. That would be the more standards way to deal with this, assuming the account is modelled as a bridge

Reply:
Right now there isn't a bridge.  A bridge would allow more than one Neato account to be configured (although I don't know if there's a need for that).  

I don't know that there's a lot to gain from that approach but I will look into it and consider changing to that.